pkg/providers/vsphere/vmprovider_vm.go

```diff
@@ -98,36 +126,104 @@ func (vs *vSphereVMProvider) CreateOrUpdateVirtualMachine(
 	}
 
 	// Set the VC UUID annotation on the VM before attempting creation or
-	// update. Among other things, the annotation facilitates differential handling
-	// of restore and fail-over operations.
+	// update. Among other things, the annotation facilitates differential
+	// handling of restore and fail-over operations.
 	if vm.Annotations == nil {
 		vm.Annotations = make(map[string]string)
 	}
-	vm.Annotations[vmopv1.ManagerID] = client.VimClient().ServiceContent.About.InstanceUuid
+	vCenterInstanceUUID := client.VimClient().ServiceContent.About.InstanceUuid
+	vm.Annotations[vmopv1.ManagerID] = vCenterInstanceUUID
 
-	vcVM, err := vs.getVM(vmCtx, client, false)
+	// Check to see if the VM can be found on the underlying platform.
+	foundVM, err := vs.getVM(vmCtx, client, false)
 	if err != nil {
 		return err
 	}
 
-	if vcVM == nil {
-		var createArgs *VMCreateArgs
+	if foundVM != nil {
+		// Mark that this is an update operation.
+		ctxop.MarkUpdate(vmCtx)
+
+		if chanErr != nil {
+			defer close(chanErr)
+		}
+		return vs.updateVirtualMachine(vmCtx, foundVM, client, nil)
+	}
+
+	// Mark that this is a create operation.
+	ctxop.MarkCreate(vmCtx)
```

Comment:
Yes, there is. It also has no impact since it is a no-op. I think I was being lazy and avoiding updating some tests that may call that more inner function. I'll check.